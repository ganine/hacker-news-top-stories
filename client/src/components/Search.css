.search {
  padding-top: 25px;
  padding-bottom: 25px;
}

.search-box {
  display: flex;
  justify-content: space-between;

  height: 30px;
  padding: 5px 10px;
  border: solid #cccccc 1px;
  border-radius: 3px;
}

.search-box form {
  width: 100%;
}

.search-box input {
  height: 100%;
  padding: 0;
  border: none;
}
.search-text {
  width: 95%;

  color: #303030;

  font-size: small;
  font-weight: bold;
}
.search-button {
  width: 5%;
  height: 58px;

  border: none;
  border-radius: 0 4px 4px 0;
}
