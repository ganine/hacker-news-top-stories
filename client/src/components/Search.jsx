import React, { useState } from 'react';

const Search = ({ search }) => {
  const [searchValue, setSearchValue] = useState('');

  const handleSearchInputChanges = e => {
    setSearchValue(e.target.value);
  };

  const resetInputField = () => {
    setSearchValue('');
  };

  const callSearchFunction = e => {
    e.preventDefault();
    search(searchValue);
    resetInputField();
  };

  return (
    <div className="search">
      <form>
        <input
          value={searchValue}
          onChange={handleSearchInputChanges}
          type="text"
          placeholder="Search stories"
        />
        <input onClick={callSearchFunction} type="submit" value="Search" />
      </form>
    </div>
  );
};

export default Search;
