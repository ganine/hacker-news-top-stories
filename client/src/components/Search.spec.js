import React from 'react';
import { render } from '@testing-library/react';
import Search from './Search';

describe('Search', () => {
  it('renders', () => {
    const { asFragment } = render(<Search />);
    expect(asFragment()).toMatchSnapshot();
  });
});
