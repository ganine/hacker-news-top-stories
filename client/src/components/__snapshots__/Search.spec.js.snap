// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search renders 1`] = `
<DocumentFragment>
  <div
    class="search"
  >
    <form>
      <input
        type="text"
        value=""
      />
      <input
        type="submit"
        value="Search"
      />
    </form>
  </div>
</DocumentFragment>
`;
